shell/platform/darwin/macos/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -568,6 +577,11 @@ - (NSTextInputContext*)inputContext {
 #pragma mark -
 #pragma mark NSTextInputClient
 
+- (void)insertTab:(id)sender {
+  // Implementing insertTab: makes AppKit send tab as command, instead of
+  // insertText with '\t'.
```

Comment:
Bit of topic, but since flutter text doesn't define tab shortcut and thus it's always handled as focus, when I wanted to insert tab I simply used this Action
```dart
class InsertTabAction extends Action<NextFocusIntent> {
  final TextEditingController controller;

  InsertTabAction(this.controller);

  void _sendIntent(Intent intent) {
    final BuildContext? primaryContext = primaryFocus?.context;
    if (primaryContext != null) {
      Actions.invoke(primaryContext, intent);
    }
  }

  @override
  Object? invoke(NextFocusIntent intent) {
    _sendIntent(ReplaceTextIntent(controller.value, '', controller.selection,
        SelectionChangedCause.keyboard));
    _sendIntent(ReplaceTextIntent(controller.value, '\t', controller.selection,
        SelectionChangedCause.keyboard));
    return null;
  }
}
```
and bound it to `NextFocusIntent` above text editor.  It's not the most obvious solution as it requires binding to `NextFocusIntent` instead of using Shortcut and custom intent, but it works even with Mac IME input.

Reply:
Can you illustrate why using `Shortcut` with `SingleActivator(LogicalKeyboardKey.tab)` and some `InsertTabIntent` wouldn't work?

Reply:
> Can you illustrate why using `SingleActivator(LogicalKeyboardKey.tab)` and some `InsertTabIntent` in a `Shortcuts` above the text editor wouldn't work?

It wouldn't. That's why I said it requires binding to `NextFocusIntent`. Tab has to be unconditionally passed to the IME (because we have no way of knowing whether the candidates popup is shown), and when we receive `insertTab:` from TextInputContext we send `NextFocusIntent` to current focus node. This is not ideal, but I don't think there's any way currently to synthesize key event and activate shortcuts, is there?